compiler/damlc/lib/DA/Cli/Damlc.hs

```diff
@@ -539,25 +539,31 @@ execCompile inputFile outputFile opts (WriteInterface writeInterface) mbIfaceDir
         createDirectoryIfMissing True $ takeDirectory outputFile
         B.writeFile outputFile $ Archive.encodeArchive bs
 
-execLint :: FilePath -> Options -> Command
-execLint inputFile opts =
+execLint :: [FilePath] -> Options -> Command
+execLint inputFiles opts =
   Command Lint (Just projectOpts) effect
   where
      projectOpts = ProjectOpts Nothing (ProjectCheck "" False)
      effect =
        withProjectRoot' projectOpts $ \relativize ->
        do
          loggerH <- getLogger opts "lint"
-         inputFile <- toNormalizedFilePath' <$> relativize inputFile
-         opts <- setDlintDataDir opts
-         withDamlIdeState opts loggerH diagnosticsLogger $ \ide -> do
-             setFilesOfInterest ide (HashSet.singleton inputFile)
-             runActionSync ide $ getDlintIdeas inputFile
-             diags <- getDiagnostics ide
-             if null diags then
-               hPutStrLn stderr "No hints"
-             else
-               exitFailure
+         inputFiles <- getInputFiles relativize inputFiles
+         forM_ inputFiles $ \inputFile -> do
```

Comment:
can we move the loop into `withDamlIdeState` and only set files of interest once to all of them? I think that should be significantly faster. You probably also want to always lint all files and only print "No hints" once (or call `exitFailure`).

Reply:
moved it to the main loop